### PR TITLE
Add warning/error result to cmd `velero backup describe`

### LINKED ISSUE #5200: error information enhancement for backup/restore
**Describe the problem/challenge you have**

Through velero cli or CR, there is high level error message, such as `PartiallyFailed` when a backup or a restore fails.
However for end users, it's still difficult to understand the actual reasons. 

For example, it may be caused by some of node network restriction to access S3 or snapshot problem for some PVC backup. 

**Describe the solution you'd like**

It's better to have some more concrete error messages which user can describe it from cli or CRs for next action.

**Anything else you would like to add:**

NA

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by allenxu404**
Thank you for contributing to Velero!

# Please add a summary of your change
- Add `backupResult`type to downloadRequest crd
- Add warning/error result to cmd `velero backup describe`
# Does your change fix a particular issue?

Fixes #5200

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by sseago** [comment]
Note that https://github.com/vmware-tanzu/velero/pull/5576 was recently merged that added backup warnings/errors to object store. From a quick look, it seems that this PR allows the newly-added results to be displayed via the backup describer. Let me know if I'm understanding the purpose here correctly and whether this PR adds onto that recently-merged result. In any case, I'll try to give this PR a more detailed look over the next week.

**Comment by allenxu404** [comment]
@sseago Yes, this PR adds warning/error data to the output of backup describer and structured describer(a new describer developed in v1.11, which is used to display output in json format, please refer to #5865 for more info) based on #5576.

**Comment by anshulahuja98** [comment]
Thank you for this contribution @allenxu404 , I had intended to add this support post #5576 
If possible, please try to share sample CLI outputs of this work. 

I will meanwhile also review the PR further

**Comment by allenxu404** [comment]
@anshulahuja98 Here are some CLI output formats:

If errors/warnings exist  && backupResults json not exists
```
velero backup describe myBackup:
Name: myBackup
...
Errors:    1
Warnings:  1
...

velero backup describe myBackup -o json:
{
    "name": "myBackup",
    "errors": {
        "count": 1
    },
    "warnings": {
        "count": 1
    },
    ...
}
```

If errors/warnings exist  && backupResults json exists
```
velero backup describe myBackup:
Name: myBackup
...
Errors:
  Velero:     <none>
  Cluster:   cluster error
  Namespaces: <none>

Warnings:
  Velero:     <none>
  Cluster:   cluster warning
  Namespaces: <none>
...

velero backup describe myBackup -o json:
{
    "name": "myBackup",
    "errors": {
        "cluster": [
            "cluster error"
        ],
        "namespace": [],
        "velero": []
    },
    "warnings": {
        "cluster": [
            "cluster warning"
        ],
        "namespace": [],
        "velero": []
    },
    ...
}
```

**Comment by allenxu404** [comment]
@sseago Do you have some time to review this PR?

**Comment by allenxu404** [comment]
> 



> not very clear to me how this will end up looking like
> 
> for example for namespaced errors are you splitting the key/value pair to represent better?

Yes,  the format is consistent with `restore describe`. Actually they share the same func.

**Comment by anshulahuja98** [comment]
I realized that later @allenxu404, and redacted that comment. Good that the function was reused.
I made some small nitpick comments, you can go through them. Otherwise the PR  LGTM.

We would need maintainers to signoff. Good to see closure on the backupresults approach in a more user friendly way.
Thank you for your contribution.

**Review by anshulahuja98 — CHANGES_REQUESTED** [review]

**Review by sseago — APPROVED** [review]

File: pkg/cmd/util/output/backup_describer.go
Comment (allenxu404) [reply pkg/cmd/util/output/backup_describer.go:571]: I might miss something but Indent here seems good to me. What problem are you seeing?

File: pkg/cmd/util/output/backup_describer.go
Comment (allenxu404) [reply pkg/cmd/util/output/backup_describer.go:580]: Make sense.

**Review by sseago — DISMISSED** [review]

**Review by ywk253100 — APPROVED** [review]

**Comment by allenxu404** [comment]
@sseago Can you give me another review? I merged the main branch and solved the conflicts.

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**
